### DISCUSSION TITLE
使用代码生成器生成代码后`localhost/admin/`页面无任何显示

### DISCUSSION THREAD

**Discussion by oknixus** [slowlyo/owl-admin#89] [Anything]
使用代码生成器，配置了一个名字叫 测速名字，Test 的功能。点击“生成代码”后，就变成一片空白了。查看控制台，显示了如下错误：
![image](https://github.com/slowlyo/owl-admin/assets/4592706/872280a4-b652-4965-a457-0691b72bfac8)
错误代码：
![image](https://github.com/slowlyo/owl-admin/assets/4592706/33c1571c-c445-4338-a029-be701f86aa6c)

一片空白：
![image](https://github.com/slowlyo/owl-admin/assets/4592706/fc2e2dd1-ba22-4bb3-9e8a-f96a16b17f57)

`php artisan admin:update --force`重新发布也不行

**Comment by slowlyo** [comment]
看下接口有没有报错

**Reply by oknixus** [reply]
接口没有报错的，status全部是0

**Reply by oknixus** [reply]
http://localhost:8006/admin-api/_settings
http://localhost:8006/admin-api/current-user
http://localhost:8006/admin-api/menus
返回的status都是0

**Reply by slowlyo** [reply]
看看返回的数据是否正常, 有没有接口返回数据为空的

**Reply by slowlyo** [reply]
检查一下菜单有没有路径重复的情况

**Reply by oknixus** [reply]
![image](https://github.com/slowlyo/owl-admin/assets/4592706/9dd47cc5-a216-4bd4-a1c7-74ec743988c0)
![image](https://github.com/slowlyo/owl-admin/assets/4592706/f8a4d7bf-d488-48a0-a642-37e2e1d75137)
![image](https://github.com/slowlyo/owl-admin/assets/4592706/e63d5959-0969-424c-ac30-995751371c6c)
没有空的

**Reply by oknixus** [reply]
从数据库中看，无重复的菜单。但是根据示例创建的菜单路径，store_admin_menus与admin_menus中的路径是一样的

**Reply by slowlyo** [reply]
![image](https://github.com/slowlyo/owl-admin/assets/94038215/50cd4557-e75d-4a44-8cc9-bc8338bdad03)

应该是这条数据的原因, 数据库把这条数据删了就好了

**Reply by oknixus** [reply]
找到原因了，代码生成器，生成的Test，在`admin_menus`中，icon字段、url字段中的都是空的，我把这条记录删掉后，就正常了

**Reply by slowlyo** [reply]
估计是因为代码生成器里面创建菜单没有做数据校验, 后续优化一下

**Reply by oknixus** [reply]
框架很不错，有用于生产环境吗？

**Reply by slowlyo** [reply]
有的

**Reply by oknixus** [reply]
老哥，想复现一下这个错误，但是又遇到了新的错误：
```
"SQLSTATE[HY000]: General error: 1364 Field 'columns' doesn't have a default value (Connection: mysql, SQL: insert into `admin_code_generators` (`title`, `updated_at`, `created_at`) values (abc, 2024-04-01 09:51:11, 2024-04-01 09:51:11))"
```
字段是有写的
![image](https://github.com/slowlyo/owl-admin/assets/4592706/7f4511ec-4421-4f1f-a913-42789da9d406)
但是就是一直报这个错误：
![image](https://github.com/slowlyo/owl-admin/assets/4592706/ac3ff758-3a9f-4fcf-8a3f-5284d50f6372)

**Reply by oknixus** [reply]
调用接口传参：
![image](https://github.com/slowlyo/owl-admin/assets/4592706/a0c8dd84-d98d-4a1d-8945-9802bfad68b8)

**Reply by slowlyo** [reply]
是最新版吗

**Reply by oknixus** [reply]
是最新版，我在`AdminService.php`中的313行加了日志：
```
 \Illuminate\Support\Facades\Log::info('参数', $data);
```
然后再次点保存的时候，就保存成功了！
开发环境使用了laravel+octane，可能和reload有关

**Reply by oknixus** [reply]
复现了这个错误了：
![image](https://github.com/slowlyo/owl-admin/assets/4592706/06336da4-317f-4ee5-8e8e-b99a4d8425ae)
![image](https://github.com/slowlyo/owl-admin/assets/4592706/d2491d92-820b-4d22-9253-0e95e750c312)
